### PR TITLE
Tutorial Refresh

### PR DESCRIPTION
**Pull request by djmitche**
Prompted by @jwhitlock going through the tutorial, I spent some time finishing up the work begun long ago with `<SiteSpecific>`.  The updated tutorial now:

 * Points people to appropriate resources on `docs.taskcluster.net` (where there's no associated root URL)
 * Points people to appropriate resources on each specific deployment, based on the root URL
 * Allows deployers to specify a good worker pool for tutorial-derived tasks.  (note that neither of our deployments has such a worker pool right now, but we can add that later for community)

### CODE REVIEW

**Review by jwhitlock — APPROVED** [review]
Thanks @djmitche, this looks like a good set of updates, and the site-specific stuff is a nice way to handle different deployments.

I found a few typos while reading, which may require re-generating search indexes. I did not try to compile and view docs locally, or review the CI output, I need to work my way up to that.

**Comment by djmitche** [comment]
@jwhitlock the "[Pages changed](https://github.com/taskcluster/taskcluster/pull/3562/checks?check_run_id=1157411101)" check in PRs gives a preview of the site.

File: ui/docs/manual/using/root-urls.mdx [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import SiteSpecific from 'taskcluster-ui/components/SiteSpecific';
+
+Every Taskcluster deployment is identified by a "Root URL", which is used to interact with the deployment.
+
+<SiteSpecific>
+The root URL for this deployment is `%root_url%`.
+</SiteSpecific>
+<SiteSpecific showIfNotSet={['root_url']}>
+The Firefox CI deployment is publicly viewable, and you are welcome to explore it.
+Its root URL is [https://firefox-ci-tc.services.mozilla.com/](https://firefox-ci-tc.services.mozilla.com/).
```
[the remark this answers is not in the document]
Comment (djmitche) [reply ui/docs/manual/using/root-urls.mdx:15]: Good point about the trailing slash!  The `<..>` syntax looks like a tag to the compiler, though, and causes errors.  But, just leaving a bare URL is sufficient!

File: ui/docs/tutorial/apis.mdx [hunk 2]
```diff
@@ -1,14 +1,44 @@
...
-examples use JavaScript (ES6) using the JavaScript [taskcluster
+examples use JavaScript and the JavaScript [taskcluster
 client](https://github.com/taskcluster/taskcluster/tree/main/clients/client) but apply to [other
-supported
-languages](/docs/manual/using/integration/libraries).
+supported languages](/docs/manual/using/integration/libraries) as well.
+
+## Authenticating to Taskcluster
+
+If you haven't already, try the "[Hello World](/docs/tutoria/hello-world)" tutorial to make
```
[the remark this answers is not in the document]
Comment (djmitche) [reply ui/docs/tutorial/apis.mdx:18]: I thought I checked all the links, but I guess I missed a few..

File: ui/docs/tutorial/finding-tasks.mdx [hunk 3]
```diff
@@ -2,12 +2,35 @@
...
 
-The "table of contents" for Taskcluster tasks is the "task index", available on the navigation menu.
+The "table of contents" for Taskcluster tasks is the "task index".
+That's a good place to start looking for real tasks.
+
+The task index browser is available in the navigation sidebar of the Taskcluster web interface.
+The precise format of the index paths depends on the tasks being run on the Taskcluster deployment you are looking at.
+
+<SiteSpecific>
+For this deployment, you can browse the root of the index [here](%root_url%/tasks/index).
```
[the remark this answers is not in the document]
Comment (djmitche) [reply ui/docs/tutorial/finding-tasks.mdx:17]: Yeah, they are pre-templating, so I'm not *too* surprised GH's struggling.  Note that these files are technically "mdx" which is some weird hybrid of JSX and Markdown.

File: ui/docs/tutorial/hello-world.mdx [hunk 4]
```diff
@@ -30,6 +37,12 @@ For example, you might see `queue:create-task:highest:tutorial/docker-worker`.
 This grants access to workerPool `tutorial/docker-worker`, also known as provisionerId `tutorial` and workerType `docker-worker`.
 If possible, choose one that is running docker-worker.
 
+<SiteSpecific>
+In this deployment, `%tutorial_worker_pool_id%` is a good choice.
+</SiteSpecific>
```
[the remark this answers is not in the document]
Comment (djmitche) [reply ui/docs/tutorial/hello-world.mdx:42]: In this case, there's nothing to say if the value isn't set, so no.

File: ui/docs/tutorial/hello-world.mdx [hunk 5]
```diff
@@ -72,3 +85,10 @@ The log view shows the output of the task, including that from downloading the d
 Scrolling all the way to the bottom, you should see the greeting output by the `echo` command.
```
[the remark this answers is not in the document]
Comment (djmitche) [reply ui/docs/tutorial/hello-world.mdx:85]: Good point!  And also, a loop that takes 600s will probably get killed with `maxRunTime: 600`, which is a cool way to illustrate that feature but probably not a good idea on the first task.  So I'll change that too.

**Pull request merged**